### PR TITLE
Use longname placeholders in the docstrings for common options

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

For "common" options, we have the option description defined in the file [`pygmt/helpers/decorators.py`](https://github.com/GenericMappingTools/pygmt/blob/c70fe880335c1ace9d9fab6ec67d31896edea117/pygmt/helpers/decorators.py#L17). Then in the module wrappers, we can use placeholders like `{R}` in the docstrings, which are automatically replaced by the long docstrings.

Currently, we use single-letter flags as the placeholders (e.g., `{R}`, `{I}`) for most common options, although there are a few exceptions (e.g., `{CPT}`, `{XY}`). It works well but sometimes may cause confusion, especially for non-common options like `-G` and `-I`.

For example, sometimes the "common" option `-G` is aliased to `fill`, but the docstring for placeholder `{G}` indicates it's aliased to `color` (e.g., https://github.com/GenericMappingTools/pygmt/issues/1617). I also remember that someone once incorrectly used the placeholder `{I}` in the docstring, but the `-I` option for the module has very different meanings (for example, `-I` means `inquiry` in some modules).

I find that using long names for these placeholders can avoid these problems and long names are also more readable than short names.

---

In this PR, I rename the placeholder `{I}` to `{spacing}` to show what the changes look like, but I plan to replace all short-name placeholders with long names (e.g., `{R}` => `{region}`).

What do you think about the changes?

References:
- https://docs.generic-mapping-tools.org/6.4/devdocs/long_options.html#common-options

### CODE REVIEW

**Comment by meghanrjones** [comment]
I agree this is an improvement. In addition to your comments, this also offers a slight benefit to users of IDEs like vscode which do not process the f-strings when displaying the documentation.

File: pygmt/tests/test_clib.py
Comment (weiji14) [inline pygmt/tests/test_clib.py:535]: These changes in the test files need to be reverted.

File: pygmt/src/basemap.py
Comment (weiji14) [inline pygmt/src/basemap.py:91]: I like the long names, but then there's a small problem with sorting the parameters. Previously we sorted by asciibetical order (e.g. ABCabc), and it made sense since the small-letter GMT parameters like `c`/`f`/`p`/`t` were rarely used. Now that we switch to the long-form, do we still go with the original GMT sorting, or change to sorting by a long-parameter's first character?

File: pygmt/helpers/decorators.py
Comment (seisman) [inline pygmt/helpers/decorators.py:403]: I think we can remove this list because the list is incomplete and there is no need to maintain it.

File: pygmt/src/basemap.py
Comment (seisman) [reply pygmt/src/basemap.py:91]: I don't have a good answer for this, but packages like numpy and pandas don't sort their parameters alphabetically. I think we also don't have to do that.

File: pygmt/helpers/decorators.py
Comment (seisman) [reply pygmt/helpers/decorators.py]: Done in 9dccaa40f5035453cbf64efd3349aa7c1d76f7ca.

File: pygmt/src/ternary.py
Comment (weiji14) [inline pygmt/src/ternary.py]:
```suggestion
    {color}
```
Fixes https://github.com/GenericMappingTools/pygmt/actions/runs/3067316855/jobs/4953471467#step:11:28
```python-traceback
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/runner/work/pygmt/pygmt/pygmt/__init__.py", line 25, in <module>
    from pygmt import datasets
  File "/home/runner/work/pygmt/pygmt/pygmt/datasets/__init__.py", line 5, in <module>
    from pygmt.datasets.earth_age import load_earth_age
  File "/home/runner/work/pygmt/pygmt/pygmt/datasets/earth_age.py", line 10, in <module>
    from pygmt.src import grdcut, which
  File "/home/runner/work/pygmt/pygmt/pygmt/src/__init__.py", line 52, in <module>
    from pygmt.src.ternary import ternary
  File "/home/runner/work/pygmt/pygmt/pygmt/src/ternary.py", line 26, in <module>
    def ternary(self, data, **kwargs):
  File "/home/runner/work/pygmt/pygmt/pygmt/helpers/decorators.py", line 481, in fmt_docstring
    module_func.__doc__ = docstring.format(**filler_text)
KeyError: 'fill'
```

Comment (weiji14) [reply pygmt/src/ternary.py]: Oops nevermind, you beat me to it :laughing:

**Review by weiji14 — APPROVED** [review]
Thanks @seisman! Was a little concerned if it's premature to find and replace the not-so-common-options like `{I|` -> `{spacing}` since we're not 100% confident that the longnames for these are considered final for GMT. I.e. that someone won't suddenly decide that `I` should be called `interval` instead of `spacing`. But should be ok since we can always change them again :slightly_smiling_face:  Certainly not a problem for the very-common-options listed in https://docs.generic-mapping-tools.org/6.4/devdocs/long_options.html#common-options such as B/R/J/U/V/X/Y and so on.

Would be good to have a second review (since this is a big PR), but I did scan through each change one by one and they seem ok.

**Comment by seisman** [comment]
Ping @maxrjones @willschlitzer @michaelgrund @yvonnefroehlich

**Comment by maxrjones** [comment]
> Ping @maxrjones @willschlitzer @michaelgrund @yvonnefroehlich

I can take a look later today or tomorrow.

**Review by yvonnefroehlich — COMMENTED** [review]
So far, all replacements appear correct to me.
The `fill`-`color` inconsistency (`pygmt.Figure.histogram` and `pygmt.Figure.ternary`) seems to be separately discussed in issue https://github.com/GenericMappingTools/pygmt/issues/1617.

File: pygmt/src/nearneighbor.py
Comment (seisman) [reply pygmt/src/nearneighbor.py:92]: > In a few functions and methods, blank lines are used between the different parameters. Currently, it appears unclear to me when and why this is done.

These blank lines are not necessary. They're there because these docstrings were written by different authors at different time.

**Review by maxrjones — APPROVED** [review]

**Pull request merged**
